src/types/subst.rs

```diff
@@ -13,6 +13,7 @@ pub(crate) struct Substitution<'a> {
     leave_lowest: usize,
     /// What to do to variables that are affected
     mapping: Mapping<'a>,
+    reg: &'a ExtensionRegistry,
```

Comment:
Could add a docstring explaining why the registry is needed for a substitution

Reply:
Yes, good point, this is better than the other places we have that comment! Can I find them....(I haven't tried to remove old comments saying the same thing)